haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs

```diff
@@ -100,15 +101,29 @@ data ExtractCtx = ExtractCtx
     }
 
 analyseTypechecked :: GhcMonad m => GhcEnv -> AnalysisOptions -> TypecheckedModule -> m XRef
-analyseTypechecked ghcEnv opts tm = do
-    let modSummary = pm_mod_summary . tm_parsed_module $ tm
+analyseTypechecked ghcEnv opts tm =
+  let
+        modSummary = pm_mod_summary . tm_parsed_module $ tm
+        renSource = tm_renamed_source tm  :: Maybe RenamedSource
+        tcSource = tm_typechecked_source tm
+        moduleTick ctx = give ctx $
+            mkModuleTick (pm_parsed_source (tm_parsed_module tm))
+                         (extractModuleName ctx (ecModule ctx))
+  in analyseTypechecked' ghcEnv opts modSummary renSource tcSource moduleTick
+
+analyseTypechecked'
+  :: GhcMonad m => GhcEnv -> AnalysisOptions
+  -> ModSummary -> Maybe RenamedSource -> LHsBinds GhcTc
+  -> (ExtractCtx -> ModuleTick)
+  -> m XRef
+analyseTypechecked' ghcEnv opts modSummary renSource tcSource fmoduleTick = do
+    let
```

Comment:
nit: here `let` is 4-indented, above 2-indented. Also leaving a new line strikes me as new. Does it actually help with indenting? I agree that the non-2-multiple indent after a `let` is generally annoying, so I'm open to best practices.